From 60fc33bd8920e3af61a1b7f0fe670269c60c2eed Mon Sep 17 00:00:00 2001
From: dineshkumar <dineshkumar@ami.com>
Date: Fri, 9 Jun 2023 10:21:47 +0000
Subject: [PATCH] Display user account Locked error message in Login

This patch will check the error code when the login API call fails,
and if the error code is 423 Locked then the WebUI will throw the
error as Reached maximum login attempts. User account locked.

Signed-off-by: dineshkumar <dineshkumar@ami.com>
---
 src/locales/en-US.json                             |  3 ++-
 src/locales/ru-RU.json                             |  3 ++-
 .../modules/Authentication/AuthenticanStore.js     | 14 +++++++++++++-
 src/views/Login/Login.vue                          | 11 ++++++++++-
 4 files changed, 27 insertions(+), 4 deletions(-)

diff --git a/src/locales/en-US.json b/src/locales/en-US.json
index c177188..e08e33b 100644
--- a/src/locales/en-US.json
+++ b/src/locales/en-US.json
@@ -686,7 +686,8 @@
     "password": "Password",
     "username": "Username",
     "alert": {
-      "message": "Invalid username or password"
+      "message": "Invalid username or password",
+      "authLockmessage": "Reached maximum login attempts. User account locked"
     }
   },
   "pageOverview": {
diff --git a/src/locales/ru-RU.json b/src/locales/ru-RU.json
index 0bfdd77..a04a041 100644
--- a/src/locales/ru-RU.json
+++ b/src/locales/ru-RU.json
@@ -605,7 +605,8 @@
     "password": "Пароль",
     "username": "Имя пользователя",
     "alert": {
-      "message": "Неверное имя пользователя или пароль"
+      "message": "Неверное имя пользователя или пароль",
+      "authLockmessage": "Достигнуто максимальное количество попыток входа. Учетная запись пользователя заблокирована."
     }
   },
   "pageOverview": {
diff --git a/src/store/modules/Authentication/AuthenticanStore.js b/src/store/modules/Authentication/AuthenticanStore.js
index 88fb54b..dc5b52c 100644
--- a/src/store/modules/Authentication/AuthenticanStore.js
+++ b/src/store/modules/Authentication/AuthenticanStore.js
@@ -7,12 +7,14 @@ const AuthenticationStore = {
   state: {
     consoleWindow: null,
     authError: false,
+    authLocked: false,
     xsrfCookie: Cookies.get('XSRF-TOKEN'),
     isAuthenticatedCookie: Cookies.get('IsAuthenticated'),
   },
   getters: {
     consoleWindow: (state) => state.consoleWindow,
     authError: (state) => state.authError,
+    authLocked: (state) => state.authLocked,
     isLoggedIn: (state) => {
       return (
         state.xsrfCookie !== undefined || state.isAuthenticatedCookie == 'true'
@@ -27,6 +29,11 @@ const AuthenticationStore = {
     },
     authError(state, authError = true) {
       state.authError = authError;
+      state.authLocked = !authError;
+    },
+    authLocked(state, authLocked = true) {
+      state.authLocked = authLocked;
+      state.authError = authLocked;
     },
     logout(state) {
       Cookies.remove('XSRF-TOKEN');
@@ -44,7 +51,11 @@ const AuthenticationStore = {
         .post('/login', { data: [username, password] })
         .then(() => commit('authSuccess'))
         .catch((error) => {
-          commit('authError');
+          if (error.response.status == 423) {
+            commit('authLocked');
+          } else {
+            commit('authError');
+          }
           throw new Error(error);
         });
     },
@@ -66,6 +77,7 @@ const AuthenticationStore = {
     },
     resetStoreState({ state }) {
       state.authError = false;
+      state.authLocked = false;
       state.xsrfCookie = Cookies.get('XSRF-TOKEN');
       state.isAuthenticatedCookie = Cookies.get('IsAuthenticated');
     },
diff --git a/src/views/Login/Login.vue b/src/views/Login/Login.vue
index 4d9fd31..8028e01 100644
--- a/src/views/Login/Login.vue
+++ b/src/views/Login/Login.vue
@@ -27,7 +27,10 @@
               variant="danger"
               :dismissible="true"
             >
-              <p id="login-error-alert">
+              <p v-if="authLocked" id="login-error-alert">
+                {{ $t('pageLogin.alert.authLockmessage') }}
+              </p>
+              <p v-else id="login-error-alert">
                 {{ $t('pageLogin.alert.message') }}
               </p>
             </alert>
@@ -140,6 +143,9 @@ export default {
     authError() {
       return this.$store.getters['authentication/authError'];
     },
+    authLocked() {
+      return this.$store.getters['authentication/authLocked'];
+    },
   },
   validations: {
     userInfo: {
@@ -249,4 +255,7 @@ export default {
   bottom: 2%;
   right: 2%;
 }
+.login-error {
+  max-width: 350px;
+}
 </style>
-- 
2.34.1

